packages/bpk-component-graphic-promo/src/BpkGraphicPromo.js

```diff
@@ -0,0 +1,202 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow strict */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { cssModules } from 'bpk-react-utils';
+import BpkText from 'bpk-component-text';
+import BpkButton from 'bpk-component-button';
+import BpkCard from 'bpk-component-card';
+
+import TEXT_ALIGN from './BpkGraphicPromoTextAlign';
+import TEXT_COLORS from './BpkGraphicPromoTextColors';
+import STYLES from './BpkGraphicPromo.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export type Props = {
+  className: ?string,
+  kicker: ?string,
+  headline: string,
+  strapline: ?string,
+  image: string,
+  sponsor: ?{
+    label: string,
+    logo: string,
+    altText: string,
+  },
+  ctaText: string,
+  onClick: () => void,
+  invertVertically: boolean,
+  textAlign: TEXT_ALIGN,
+  textColor: TEXT_COLORS,
+};
+
+const constructAriaLabel = ({
+  ctaText,
+  headline,
+  kicker,
+  sponsor,
+  strapline,
+}: Props) => {
+  const text = [];
+  const addText = (value) => value && text.push(value);
+
+  if (sponsor) {
+    addText(sponsor.label);
+    addText(sponsor.altText);
+  } else {
+    addText(kicker);
+  }
+  addText(headline);
+  addText(strapline);
+  addText(ctaText);
+
+  return text.join('. ');
+};
```

Comment:
Should this be covered in a test?

Reply:
The tests are pretty basic and using snapshots. This function is covered because we take a snapshot for every variant of the component (sponsored, with/without kicker and strapline etc.). Ideally, I would have tested this separately and not in a snapshot, but I tried to stay consistent with the rest of the codebase. 🙂